web/src/popover_menus.js

```diff
@@ -192,6 +208,84 @@ export function toggle_message_actions_menu(message) {
     return true;
 }
 
+const send_later_hours = {
+    in_one_hour: {
+        text: $t({defaultMessage: "1 hour"}),
+        hours: 1,
+    },
+    in_two_hours: {
+        text: $t({defaultMessage: "2 hours"}),
+        hours: 2,
+    },
+    in_four_hours: {
+        text: $t({defaultMessage: "4 hours"}),
+        hours: 4,
+    },
+};
+
+const send_later_tomorrow = {
+    tomorrow_nine_am: {
+        text: $t({defaultMessage: "Tomorrow 9:00 AM"}),
+        time: "9:00 am",
+    },
+    tomorrow_two_pm: {
+        text: $t({defaultMessage: "Tomorrow 2:00 PM "}),
+        time: "2:00 pm",
+    },
+};
+
+const send_later_days_and_weeks = {
+    in_one_day: {
+        text: $t({defaultMessage: "1 day"}),
+        days: 1,
+    },
+    in_two_days: {
+        text: $t({defaultMessage: "2 days"}),
+        days: 2,
+    },
+    in_one_week: {
+        text: $t({defaultMessage: "1 week"}),
+        days: 7,
+    },
+    in_two_weeks: {
+        text: $t({defaultMessage: "2 weeks"}),
+        days: 14,
+    },
+    in_one_month: {
+        text: $t({defaultMessage: "1 month"}),
+        days: 30,
+    },
+};
+
+const send_later_custom = {
+    text: $t({defaultMessage: "Custom"}),
+};
+
+function reset_compose_scheduling_state_on_success() {
+    $("#compose-textarea").prop("disabled", false);
+    $("#schedule-confirm").hide();
+    $("#schedule-confirm")[0]._tippy.destroy();
+    reset_selected_schedule_time();
+    $("#compose-send-button").show();
+    compose.clear_compose_box();
+}
+
+export function schedule_message_to_custom_date() {
+    const request = compose.create_message_object();
+    request.content = `/schedule ${selected_send_later_time}\n` + request.content;
+    // If this is an edit request `scheduled_message_id` will be defined.
+    let scheduled_message_id;
+    if ($("#compose-textarea").attr("data-scheduled-message-id")) {
+        scheduled_message_id = $("#compose-textarea").attr("data-scheduled-message-id");
+        $("#compose-textarea").removeAttr("data-scheduled-message-id");
+    }
+    reminder.schedule_message(
+        request,
+        reset_compose_scheduling_state_on_success,
+        scheduled_message_id,
+    );
+}
```

Comment:
These couple functions feel like they should live in `reminder.js`, a new `scheduled_messages.js`, or `compose.js`, rather than in the popovers module.

Reply:
Okay, moved to `compose` since we were already moving all the sending logic to `compose.finish`.